BBHX_Phenom.py

```diff
@@ -55,8 +62,15 @@ def interpolated_tf(m1, m2):
     tf_track = interp1d(t_array, freq_array)
     return tf_track
 
-def bbhx_fd(ifos=None, run_phenomd=True,
-            ref_frame='LISA', sample_points=None, **params):
+def waveform_setup(**kwargs):
+    if kwargs['approximant'] == "BBHX_PhenomD":
+        kwargs['mode_array'] = [(2, 2)]
+        return _bbhx_fd(**kwargs)
+    elif kwargs['approximant'] == "BBHX_PhenomHM":
+        return _bbhx_fd(run_phenomd=False, **kwargs)
```

Comment:
If I'm following the logic correctly, I think this means there isn't a default `mode_array` for PhenomHM. In most cases, I think users would assume setting `approximant='BBHX_PhenomHM'` would enable all the modes, so I'd suggest adding a default. Maybe something like:

```suggestion
    elif kwargs['approximant'] == "BBHX_PhenomHM":
        if 'mode_array' not in kwargs:
            kwargs['mode_array'] = [(2, 2), (2, 1), (3, 3), (3, 2), (4, 4), (4, 3)]
        return _bbhx_fd(run_phenomd=False, **kwargs)
```

Reply:
@mj-will  If no modes are given to PhenomHM, by default, BBHx returns all available modes. So that is why we don't enforce here what the modes should be because you want to be able to call individual modes from PhenomHM as well as potentially a few modes at once.

Reply:
Just added a raise exception if the mode_arrary is not given when using PhenomHM

Reply:
I'm not sure about this change, it means the user always has to specify the mode array. Normally, if I ask for an approximant, I'd expect it to have some sort of default for the modes (which I think is what you had before)

Reply:
Yeah I see what you mean. My logic here was making sure that the user always knows what modes they're getting back by forcing them to specify the mode content. In the multi_signal model, I specify for each submodel what mode_array it is. If somebody doesn't do this they could potentially run a submodel with all default modes and it'll mess the inference up.

Reply:
Maybe there should be a message given in the log to say "Modes set to default values of 'blah'" ? Then they can trace it back easily?

Reply:
I get that in the context of the multi_signal model, but I think in general you'd expect a default for normal use, like when you use the standard IMRPhenomXPHM for ground-based inference.

I think you could have a onetime log statement/warning but I would personally avoid something like that being printed every time. Though I don't know what the default approach is for these sorts of things in pycbc, so that might be a question for @ahnitz or @spxiwh

Reply:
@mj-will I've added your requested change!